File: sample/exampleui/src/main/java/imui/jiguang/cn/imuisample/fragment/common/ShadowFragment.java
package imui.jiguang.cn.imuisample.fragment.common;


import android.app.Fragment;
import android.content.Context;
import android.os.Bundle;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import imui.jiguang.cn.imuisample.R;

public class ShadowFragment extends Fragment {

    private  ShadowScreen shadow;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.shadow, container, false);
        shadow= (ShadowScreen) getActivity();

        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                shadow.shadowOnclick();
            }
        });
        return view;
    }


    public interface  ShadowScreen{
        public void shadowOnclick();
    }

}

File: app/src/main/java/com/example/sk_android/mvp/view/fragment/common/ShadowFragment.kt
package com.example.sk_android.mvp.view.fragment.common

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.*
import com.example.sk_android.R
import org.jetbrains.anko.*
import org.jetbrains.anko.support.v4.UI
import android.content.Context

class ShadowFragment : Fragment() {


    private var mContext: Context? = null
    private lateinit var shadowClick:ShadowClick

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = activity

    }

    companion object {
        fun newInstance(): ShadowFragment {
            val fragment = ShadowFragment()
            return fragment
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        var fragmentView=createView()
        shadowClick =  activity as ShadowClick
        return fragmentView
    }

    fun createView(): View {

        return UI {
            linearLayout {
                verticalLayout()  {
                    setOnClickListener(object :View.OnClickListener{
                        override fun onClick(v: View?) {
                            shadowClick.shadowClicked()
                        }
                    })
                    backgroundColorResource=R.color.shadowColor
                }.lparams(width = matchParent, height =matchParent){

                }
            }
        }.view
    }

    public interface ShadowClick {

        fun shadowClicked( )
    }


}

